features/FEATURE_UVISOR/includes/uvisor/api/inc/uvisor_deprecation.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2016, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef __UVISOR_DEPRECATION_H__
+#define __UVISOR_DEPRECATION_H__
+
+#if defined(UVISOR_PRESENT) && UVISOR_PRESENT == 1
+#warning "---- WARNING: You are using FEATURE_UVISOR which is deprecated since mbed-os-5.9!! ----"
```

Comment:
Perhaps something like:

"Warning: In 5.10, Arm Mbed uVisor is being deprecated in favor of our upcoming security features. We strongly encourage you to use these new security features instead."

I'd like to link to mention and link to PSA or PSA crypto, but I don't know if linking to a private repo makes sense.

@ndevillard @ChiefBureaucraticOfficer Please advise.

Reply:
After talking with @ChiefBureaucraticOfficer we suggest "Warning: You are using FEATURE_UVISOR, which is unsupported as of Mbed OS 5.9."

@ndevillard @simonfordarm What are your thoughts on this phrasing?